src/providers/Theme.tsx

```diff
@@ -0,0 +1,55 @@
+import type { Theme } from "@material-ui/core/styles";
+import {createContext, ReactChild, useEffect, useState} from "react"
+import { lightTheme, darkTheme } from "../theme"
+
+interface ContextInterface {
+    themeMode: ThemeMode
+    theme: Theme
+    toggleThemeMode: () => void
+}
+
+const initialValues: ContextInterface = {
+    themeMode: 'light',
+    theme: lightTheme,
+    toggleThemeMode: () => {}
+}
+
+const Context = createContext<ContextInterface>(initialValues);
+
+interface Props {
+    children: ReactChild
+}
+
+function Provider({children}: Props) {
+    const [themeMode, setThemeMode] = useState<ThemeMode>(initialValues.themeMode)
+    const theme = themeMode === 'light' ? lightTheme : darkTheme
+    const toggleThemeMode = () => {
+        const newThemeMode = themeMode === 'light' ? 'dark' : 'light'
+        setThemeMode(newThemeMode)
+        localStorage.setItem('theme', newThemeMode)
+    }
+
+    // Detect previously store theme
+    useEffect(() => {
+        const newThemeMode: ThemeMode = localStorage.getItem('theme') === 'light' ? 'light' : 'dark'
+
+        setThemeMode(newThemeMode)
+
+        window?.matchMedia('(prefers-color-scheme: dark)')?.addEventListener('change', e => {
```

Comment:
You pass an anonymous function to the `addEventListener` and another (but functionally same) anonymous function to `removeEventListener`. This is undefined (browser-dependent) behavior and usually it's better to extract the anonymous function and assign it to a variable, and pass the variable to the `addEventListener` and `removeEventListener` calls.